provisioning/baremetal_secrets_test.go

```diff
@@ -0,0 +1,151 @@
+
+			kubeClient := fakekube.NewSimpleClientset(nil...)
+
+			if tc.secretError != nil {
+				kubeClient.Fake.PrependReactor("get", "secrets", func(action faketesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1.Secret{}, tc.secretError
+				})
+			}
+
+			err := CreateMariadbPasswordSecret(kubeClient.CoreV1(), testNamespace)
+			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				secret, _ := kubeClient.Tracker().Get(secretsResource, testNamespace, "metal3-mariadb-password")
+				assert.NotEmpty(t, secret.(*v1.Secret).StringData[baremetalSecretKey])
+			}
+		})
+	}
+}
+
+// Testing the case where the Mariadb password already exists
+// First we create a Mariadb Password using the method being tested.
+// Then we attenpt to create it again by calling the method again.
+// Instead of creating a new one, it should return the pre-existing secret
+func TestDuplicateMariadbPasswordCreation(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	client := kubeClient.CoreV1()
+
+	// First create a mariadb password secret
+	if err := CreateMariadbPasswordSecret(kubeClient.CoreV1(), testNamespace); err != nil {
+		t.Fatalf("Failed to create first Mariadb password. %s ", err)
+	}
+	// Read and get Mariadb password from Secret just created.
+	oldMariadbPassword, err := client.Secrets(testNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Failure getting the first Mariadb password that just got created.")
+	}
+	oldPassword, ok := oldMariadbPassword.StringData[baremetalSecretKey]
+	if !ok || oldPassword == "" {
+		t.Fatal("Failure reading first Mariadb password from Secret.")
+	}
+	// The pasword definitely exists. Try creating again.
+	if err := CreateMariadbPasswordSecret(kubeClient.CoreV1(), testNamespace); err != nil {
+		t.Fatal("Failure creating second Mariadb password.")
+	}
+	newMariadbPassword, err := client.Secrets(testNamespace).Get(context.Background(), baremetalSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Failure getting the second Mariadb password.")
+	}
+	newPassword, ok := newMariadbPassword.StringData[baremetalSecretKey]
+	if !ok || newPassword == "" {
+		t.Fatal("Failure reading second Mariadb password from Secret.")
+	}
+	if oldPassword != newPassword {
+		t.Fatalf("Both passwords do not match.")
+	} else {
+		t.Logf("First Mariadb password is being preserved over re-creation as expected.")
+	}
+}
+
+func TestCreateIronicPasswordSecret(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	client := kubeClient.CoreV1()
+
+	err := CreateIronicPasswordSecret(client, testNamespace)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+	// Check if Ironic secret exits
+	secret, err := client.Secrets(testNamespace).Get(context.Background(), ironicSecretName, metav1.GetOptions{})
```

Comment:
I think it would be better to use ` kubeClient.Tracker().Get` in test code to verify that the mock registered correctly the new stored secret

Reply:
Done.